waiter/test/waiter/handler_test.clj

```diff
@@ -846,6 +846,22 @@
         (is (get (json/read-str body) "success?"))
         (is (<= update-delay (- timeout 50) (- end-time start-time) (+ timeout 50)))))
 
+    (testing (str handler-name ":success-with-update")
+      (let [fallback-state-atom (atom {:available-service-ids #{"s1"}})
+            timeout 10000
+            update-delay 2000
+            request-query (assoc request :query-string (str "timeout=" timeout))
+            start-time (System/currentTimeMillis)
```

Comment:
Start time should be after the go block.
This test is sensitive to scheduling delays, is there some other way we can test the running time?

Reply:
I couldn't find a better way for time measurement. Because the timeout is much larger than the update-delay by 8 seconds, the ending assertion `(<= update-delay (- end-time start-time) timeout)` should still hold in the scenario that the update thread starts late or sleeps over 2 seconds. 

Also, I don't think there would be a scenario where the `service-await-handler` could finish before 2 seconds (violating inequality: update-delay <= end-time - start-time) unless the thread can sleep for less than 2 seconds?